service/src/integrationTest/kotlin/fi/espoo/evaka/messaging/MessageIntegrationTest.kt

```diff
@@ -1156,6 +1204,150 @@ class MessageIntegrationTest : FullApplicationTest(resetDbBeforeEach = true) {
         }
     }
 
+    @Test
+    fun `recipient list can be filtered`() {
+        postNewThread(
+            title = "Vappu",
+            message = "Vappuna paistaa aurinko",
+            messageType = MessageType.BULLETIN,
+            sender = messagerAccount,
+            recipients =
+                listOf(
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_1.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_3.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_4.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_6.id),
+                    MessageRecipient(MessageRecipientType.CHILD, testChild_8.id),
+                ),
+            filters = MessageController.PostMessageFilters(yearsOfBirth = listOf(2017)),
+            user = messager,
+            now = sendTime
+        )
+
+        db.read {
+            assertEquals(
+                2,
+                @Suppress("DEPRECATION")
+                it.createQuery("SELECT COUNT(id) FROM message_recipients").exactlyOne<Int>()
```

Comment:
Use the new API instead of the deprecated old one: `it.createQuery { sql(...) }`